src/shortestpaths/dijkstra.jl

```diff
@@ -169,3 +169,115 @@ function dijkstra_shortest_paths(
         g, [src;], distmx; allpaths=allpaths, trackvertices=trackvertices, maxdist=maxdist
     )
 end
+
+function relax(u::U, 
+               v::U, 
+               distmx::AbstractMatrix{T}, 
+               dists::Vector{T}, 
+               parents::Vector{U}, 
+               visited::Vector{Bool}, 
+               Q::PriorityQueue{U,T}
+) where {T<:Real} where {U<:Integer}
+     alt = dists[u] + distmx[u, v]
+
+     if !visited[v]
+         visited[v] = true
+         dists[v] = alt
+         parents[v] = u
+         Q[v] = alt
+     elseif alt < dists[v]
+         dists[v] = alt
+         parents[v] = u
+         Q[v] = alt
+     end
+end
+
+"""
+    bidijkstra_shortest_paths(g, src, dst, distmx=weights(g));
+
+Perform [Bidirectional Dijkstra's algorithm](https://www.homepages.ucl.ac.uk/~ucahmto/math/2020/05/30/bidirectional-dijkstra.html)
+on a graph, computing the shortest path between `src` and `dst`.
+
+# Examples
+```jldoctest
+julia> using Graphs
+
+julia> bidijkstra_shortest_path(cycle_graph(5), 1, 4);
+3-element Vector{Int64}:
+ 1
+ 5
+ 4
+
+julia> bidijkstra_shortest_path(path_graph(5), 1, 4);
+4-element Vector{Int64}:
+ 1
+ 2
+ 3
+ 4
+```
+"""
+function bidijkstra_shortest_path(
+    g::AbstractGraph,
+    src::U,
+    dst::U,
+    distmx::AbstractMatrix{T}=weights(g)
+) where {T<:Real} where {U<:Integer}
+    if src == dst
+        return Int64[]
+    end
+    # keep weight of the best seen path and the midpoint vertex
+    mu, mid_v = typemax(T), -1
+
+    nvg = nv(g)
+    dists_f, dists_b= fill(typemax(T), nvg), fill(typemax(T), nvg)
+
+    parents_f, parents_b= zeros(U, nvg), zeros(U, nvg)
+
+    visited_f, visited_b = zeros(Bool, nvg),zeros(Bool, nvg)
+
+    preds_f, preds_b = fill(Vector{U}(), nvg), fill(Vector{U}(), nvg)
+
+    Qf, Qb = PriorityQueue{U,T}(), PriorityQueue{U,T}()
+
+    dists_f[src] = zero(T)
+    visited_f[src] = true
+    Qf[src] = zero(T)
+
+    dists_b[dst] = zero(T)
+    visited_b[dst] = true
+    Qb[dst] = zero(T)
+
+    while !isempty(Qf) && !isempty(Qb)
+        uf, ub = dequeue!(Qf), dequeue!(Qb)
+
+        for v in outneighbors(g, uf)
+            relax(uf, v, distmx, dists_f, parents_f, visited_f, Qf)
```

Comment:
This relax function clarifies things, would it be hard to add it to the original `dijkstra` too?

Reply:
I was hoping you would notice =p I'll make the changes

Reply:
I have a new version of the `relax` function  that should work for both the `dijkstra_shortest_paths` and `bidijkstra_shortest_path` but it is a bit clunky. I am wondering; why the need for the `parents` and `preds` structures ? wouldn't the list of predecessors be enough ? 
A cosmetic note: should we rename the function the `bidijkstra_shortest_paths` if it should be able to deal with multiple paths ?